src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -1334,8 +1339,9 @@ void CodeGen::genStructReturn(GenTree* treeNode)
             }
         }
     }
-#elif defined(_TARGET_AMD64_)
-    genConsumeReg(op1); // __m128 should be returned in XMM0
+#elif defined(_TARGET_AMD64_) && defined(_TARGET_WINDOWS_)
+    // TYP_SIMD16 should be returned in XMM0
```

Comment:
It looks like, for x64 Windows, we need to also return `TYP_SIM32` in `YMM0`: https://godbolt.org/z/3zSHW7
* The x64 calling convention docs aren't explicit on this point, but the disassembly shows this is the case: <https://docs.microsoft.com/en-us/cpp/build/x64-calling-convention?view=vs-2019#return-values>

Reply:
Thanks for the feedback. From the compiler explorer link you sent, it is likely that you're right. I have sent an email to the C++ team to ask for confirmation.

If the behavior is confirmed, it makes sense to fix both cases since both of these rules will probably require similar changes at similar places.

The work to track handling this is marked as a checkbox at the beginning of the PR.